connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala

```diff
@@ -109,44 +109,131 @@ class RelationalGroupedDataset protected[sql] (
     agg(exprs.asScala.toMap)
   }
 
-  private[this] def strToExpr(expr: String, inputExpr: proto.Expression): proto.Expression = {
+  private[this] def strToExpr(expr: String, columnName: String): proto.Expression = {
     val builder = proto.Expression.newBuilder()
 
     expr.toLowerCase(Locale.ROOT) match {
       // We special handle a few cases that have alias that are not in function registry.
       case "avg" | "average" | "mean" =>
-        builder.getUnresolvedFunctionBuilder
-          .setFunctionName("avg")
-          .addArguments(inputExpr)
-          .setIsDistinct(false)
+        functions.avg(columnName)
```

Comment:
You currently don't return this function, but the result of builder.build(). If you do, it should be `functions.avg(columnName).expr`.

Reply:
I think I did right replacement and hit a proto -> plan test generation failure. 

I am planing look into that separately. I am gonna need some time to learn how to debug `org.apache.spark.sql.connect.ProtoToParsedPlanTestSuite`